### ISSUE TITLE
production=false flag doesn't work as expected

### ISSUE THREAD

**Issue by alex-pex** [yarnpkg/yarn#4641]
**Do you want to request a *feature* or report a *bug*?**
It is a bug

**What is the current behavior?**
```NODE_ENV=production yarn install --production=false``` doesn't install devDependencies. It's like the flag is ignored.

**If the current behavior is a bug, please provide the steps to reproduce.**
You can use this package.json
```
{
  "private": true,
  "engines": {
    "node": ">=6.9.0 <9.0.0",
    "yarn": "^1.0.0"
  },
  "devDependencies": {
    "aaa": "^0.0.2"
  }
}
```

**What is the expected behavior?**
I should have the "aaa" package in node_modules. Like what I get when I'm doing ```NODE_ENV=development yarn install```.

https://yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-production-true-false

**Please mention your node.js, yarn and operating system version.**
I'm on Windows with Git Bash.

```
$ node -v
v8.5.0

$ npm -v
5.3.0

$ yarn -v
1.1.0
```

**Comment by alexbrazier** [comment]
This is also an issue for me on macOS

yarn `v1.1.0`

**Comment by alexbrazier** [comment]
Looks like this has been resolved, and will be in the next release - 8580a5831e6239353d0ffba1c4df678eaa31161e

Should also fix #4646

Related: #4557

**Comment by BYK** [comment]
Closing since this is fixed on master.
